pkg/agent/cniserver/ipam/antrea_ipam.go

```diff
@@ -179,30 +186,34 @@ func (d *AntreaIPAM) Check(args *invoke.Args, k8sArgs *argtypes.K8sArgs, network
 // Antrea IPAM annotation for the resource (only Namespace annotation is supported as
 // of today). If annotation is not present, or annotated IP Pool not found, the driver
 // will not own the request and fall back to next IPAM driver.
-func (d *AntreaIPAM) owns(k8sArgs *argtypes.K8sArgs) (bool, *poolallocator.IPPoolAllocator, error) {
+func (d *AntreaIPAM) owns(k8sArgs *argtypes.K8sArgs) (bool, *poolallocator.IPPoolAllocator, []net.IP, error) {
 	if d.controller == nil {
 		klog.Warningf("Antrea IPAM driver failed to initialize due to inconsistent configuration. Falling back to default IPAM")
-		return false, nil, nil
+		return false, nil, nil, nil
 	}
 
 	// As of today, only Namespace annotation is supported
 	// In future, Deployment, Statefulset and Pod annotations will be
 	// supported as well
 	namespace := string(k8sArgs.K8S_POD_NAMESPACE)
-	klog.V(2).InfoS("Inspecting IPAM annotation", "namespace", namespace)
-	poolNames := d.controller.getIPPoolsByNamespace(namespace)
-	if len(poolNames) < 1 {
-		return false, nil, nil
+	podName := string(k8sArgs.K8S_POD_NAME)
+	klog.V(2).InfoS("Inspecting IPAM annotation", "namespace", namespace, "pod", podName)
+	poolNames, ips, err := d.controller.getIPPoolsByPod(namespace, podName)
```

Comment:
Should not we check at most one v4 and one v6 IPs? And check whether v4/v6 is enabled (did we check IP pools for this)?

Reply:
Yes, but IPv6 is not supported now, thus we only support one pool and one ip in current version.
https://github.com/antrea-io/antrea/pull/3042/files/d1fc8215367d46117536a56fd891fdf9b7a26170#diff-b0835971aa431617cb107905707afd58dc43d6a495d3b17bc7c8965bfbfc8578R205

Reply:
Got it. I did not notice pool must be provided too to specify an IP.